src/main/groovy/org/shipkit/internal/gradle/CloneGitRepositoryTask.java

```diff
@@ -25,37 +31,92 @@
 
     private static final Logger LOG = Logging.getLogger(CloneGitRepositoryTask.class);
 
-    private String repository;
+    private String repositoryUrl;
     private File targetDir;
+    private int numberOfCommitsToClone;
 
     @TaskAction
     public void cloneRepository() {
-        LOG.lifecycle("  Cloning repository {}\n    into {}", repository, targetDir);
+        LOG.lifecycle("  Cloning repository {}\n    into {}", repositoryUrl, targetDir);
         getProject().getBuildDir().mkdirs();    // build dir can be not created yet
         ProcessRunner processRunner = Exec.getProcessRunner(getProject().getBuildDir());
-        processRunner.run("git", "clone", repository, targetDir.getAbsolutePath());
+        processRunner.run(getCloneCommand());
+    }
+
+    @ExposedForTesting
+    List<String> getCloneCommand() {
+        List<String> result = new ArrayList<String>();
+        result.add("git");
+        result.add("clone");
+        if(numberOfCommitsToClone != 0) {
```

Comment:
Good idea. Makes the task more robust. Shall we use some sensible default for cloneDepth (like 50)?